### PR TITLE
Transformers update

### PR DESCRIPTION
**Pull request by clamor-s**

### CODE REVIEW

File: arch/arm/boot/dts/tegra30-asus-transformer-common.dtsi [hunk 1]
```diff
@@ -1390,7 +1390,6 @@
 		regulator-boot-on;
 		gpio = <&pmic 6 GPIO_ACTIVE_HIGH>;
 		enable-active-high;
-		vin-supply = <&vdd_5v0_bat>;
```
Comment (digetx) [inline arch/arm/boot/dts/tegra30-asus-transformer-common.dtsi:1393]: Is anything wrong here?

**Comment by digetx** [comment]
@davidhildenbrand Looks like it tries to make a release tag here.

File: arch/arm/boot/dts/tegra30-asus-transformer-common.dtsi (hunk 1 above)
Comment (clamor-s) [reply arch/arm/boot/dts/tegra30-asus-transformer-common.dtsi:1393]: 3.3v according to schematis is not supplied by anything and starts same as 5v bat and they are not connected

Comment (digetx) [reply arch/arm/boot/dts/tegra30-asus-transformer-common.dtsi:1393]: The TF700T pdf says that the 3VSUS is sourced from the battery supply. I assume this should be the same for all transformers.

Comment (clamor-s) [reply arch/arm/boot/dts/tegra30-asus-transformer-common.dtsi:1393]: [https://github.com/clamor-s/files/blob/main/ASUS%20Pegatron%20TF201X%20TF300T_RG.pdf](second page)

Comment (clamor-s) [reply arch/arm/boot/dts/tegra30-asus-transformer-common.dtsi:1393]: yes, but 5v bat is not battery supply. 5v sys and 5v sby descends from it, but not 3.3v

Comment (digetx) [reply arch/arm/boot/dts/tegra30-asus-transformer-common.dtsi:1393]: The schematics clearly shows that 3.3 and 5 are generated from AC_BAT_SYS by the TPS charger IC.

Comment (clamor-s) [reply arch/arm/boot/dts/tegra30-asus-transformer-common.dtsi:1393]: ac-bat-sys = vdd-ac-bat?

Comment (digetx) [reply arch/arm/boot/dts/tegra30-asus-transformer-common.dtsi:1393]: yes

Comment (clamor-s) [reply arch/arm/boot/dts/tegra30-asus-transformer-common.dtsi:1393]: Ok, will update.

File: arch/arm/boot/dts/tegra30-asus-transformer-common.dtsi [hunk 2]
```diff
@@ -1230,9 +1230,9 @@
 				};
 
 				vdd2_reg: vdd2 {
-					regulator-name = "vddio_ddr";
-					regulator-min-microvolt = <600000>;
-					regulator-max-microvolt = <1500000>;
+					regulator-name = "vdd_1v2_gen";
+					regulator-min-microvolt = <1200000>;
+					regulator-max-microvolt = <1200000>;
```
Comment (digetx) [inline arch/arm/boot/dts/tegra30-asus-transformer-common.dtsi:1235]: Could you please tell where from you got these names and values?

Comment (clamor-s) [reply arch/arm/boot/dts/tegra30-asus-transformer-common.dtsi:1235]: https://github.com/grate-driver/linux/blob/master/arch/arm/boot/dts/tegra30-asus-nexus7-grouper-ti-pmic.dtsi#L44

Comment (digetx) [reply arch/arm/boot/dts/tegra30-asus-transformer-common.dtsi:1235]: Have you checked it on all transformer devices?

Comment (clamor-s) [reply arch/arm/boot/dts/tegra30-asus-transformer-common.dtsi:1235]: Yes, tf201/tf300t/tf700t both native and kexec. Furthermore, thos values can be logged from downstream is /sys/

Comment (maximschwalm) [reply arch/arm/boot/dts/tegra30-asus-transformer-common.dtsi:1235]: Weren't the names correct?

Comment (clamor-s) [reply arch/arm/boot/dts/tegra30-asus-transformer-common.dtsi:1235]: @digetx is value 8 legit for ti,regulator-ext-sleep-control

**Comment by clamor-s** [comment]
@digetx looks like tf701t dock actually is a i2c device and CAN be described as asus,dock-ec. I;d ask you to revert part of changes I've proposed. Me and jenneron are currently working on proper solution that does not need specific per-device data definitions.

File: drivers/mfd/asus-ec.c [hunk 3]
```diff
@@ -515,9 +480,12 @@ static int asus_ec_remove(struct i2c_client *client)
 	struct asus_ec_data *priv = i2c_get_clientdata(client);
 
 	mfd_remove_devices(&priv->self->dev);
+
 	sysfs_remove_link(&client->dev.kobj, "dockram");
 	sysfs_remove_group(&client->dev.kobj, &asus_ec_attr_group);
 
+	priv->logging_disabled = false;
```
Comment (digetx) [inline drivers/mfd/asus-ec.c:487]: The driver is removed, logging_disabled should already be false and it won't do anything useful anyways. This line is unneeded.

Comment (clamor-s) [reply drivers/mfd/asus-ec.c:487]: Ok, I doubted on this. Thanks, will remove.

**Review by digetx — COMMENTED** [review]
TF101 would never boot previously. if it required firmware. Downstream kernels may have TF code, but it should be disabled by default. T20 consumer devices don't use TF in general.

File: arch/arm/boot/dts/tegra30-asus-transformer-common.dtsi (hunk 2 above)
Comment (digetx) [reply arch/arm/boot/dts/tegra30-asus-transformer-common.dtsi:1235]: Likely that the names were correct and N7 has incorrect names, but I'm not 100% sure.

**Comment by clamor-s** [comment]
> TF101 would never boot previously. if it required firmware. Downstream kernels may have TF code, but it should be disabled by default. T20 consumer devices don't use TF in general.

Agree, there are some thoughts about booting with android bootloader. TF101G does not have nvflash ability.

May this work? https://github.com/torvalds/linux/blob/master/arch/arm/boot/dts/tegra114-tn7.dts#L14-L16
Jenneron proposed to check, but none of TF101 users provided addresses needed.

**Comment by digetx** [comment]
Missing initrd shouldn't prevent kernel from booting. The problem that we had on N7 is that bootloader overwrites part of kernel with initrd if kernel image is too big. But it could be more complicated, like an odd h/w configuration left after bootloader. It's possible to debug kernel booting blindly by inserting PMC reboot call for finding the place in the code that causes the hang.

Enabling firmware for TF101 should break it entirely, please don't do it. I suppose that the TF patch wasn't tested, correct?

**Comment by clamor-s** [comment]
Actually it was and it worked. At least it did not brake anything. I'll call tf101 users in pmos on transformer channel for more info.

File: drivers/power/supply/asus-ec-battery.c [hunk 4]
```diff
@@ -237,14 +236,14 @@ static int asusec_battery_probe(struct platform_device *pdev)
...
-	priv->ec_addr = pdata->ec_addr;
+	priv->ec_addr = 0x14;
 	priv->batt_data_ts = jiffies - 1;
 	priv->last_state = POWER_SUPPLY_STATUS_UNKNOWN;
 
 	cfg.of_node = pdev->dev.parent->of_node;
 	cfg.drv_data = priv;
 
-	if (pdata->is_pad)
+	if (!of_property_match_string(cfg.of_node, "compatible", "asus,pad-ec"))
```
Comment (digetx) [inline drivers/power/supply/asus-ec-battery.c:246]: Couldn't `of_device_is_compatible()` be use here?

**Comment by digetx** [comment]
Picasso doesn't have such problem.

File: drivers/power/supply/asus-ec-battery.c (hunk 4 above)
Comment (digetx) [reply drivers/power/supply/asus-ec-battery.c:246]: I'll change it after merging. The rest looks okay.

**Comment by digetx** [comment]
Thanks!

**Pull request merged**